### ISSUE TITLE
TarWriter writes badly formatted size field for a zero sized regular file

### ISSUE THREAD

**Issue by sobczyk** [dotnet/runtime#95354]
### Description

TarWriter writes badly formatted size field for a zero sized regular file

### Reproduction Steps

```cs
using System;
using System.IO;
using System.Linq;
using System.Formats.Tar;

MemoryStream emptyData = new(0);
MemoryStream output = new();
TarWriter archive = new(output, TarEntryFormat.Pax);
PaxTarEntry te = new(TarEntryType.RegularFile, "zero_size")
{DataStream = emptyData};
archive.WriteEntry(te);
// get the size[12] field, after two unrelated blocks[512] and name[100], mode[8], uid[8], gid[8]
var offset = 2*512+100+3*8;
var sizeBuffer=output.GetBuffer()[offset..(offset+12)];
// it's "00000000000\0"
byte[] expected = new byte[] {0x30,0x30,0x30,0x30,0x30,0x30,0x30,0x30,0x30,0x30,0x30,0};
Console.WriteLine($"Proper size field: {sizeBuffer.SequenceEqual(expected)}");
```

### Expected behavior

get all "0" in field (0x32) followed by "\0"

### Actual behavior

get binary zero

### Regression?

No

### Known Workarounds

None

### Configuration

Affected:  (any with TarWriter)
* .net 7
* .net 8

**Comment by sobczyk** [comment]
This breaks unpacking by tar-rs https://github.com/alexcrichton/tar-rs/issues/347

**Comment by danmoseley** [comment]
Thanks for the report. Any interest in offering a PR?

**Comment by sobczyk** [comment]
Ok, I'll try to fix this

**Comment by danmoseley** [comment]
It is unusual that any tests fail in main. It really would only happen if they were outer loop as those aren't validated on every PR.

**Comment by ericstj** [comment]
@sobczyk did you have a PR for this?  Our window for 9.0 is closing soon, but we still have time to look at a PR if available in the next week or so.

**Comment by sobczyk** [comment]
@ericstj the change is basically a one liner, the problem is that the unit tests are a bit messy
I'll provide the fix today or tomorrow, and will try to do some UT, but that looked more difficult than the fix

**Comment by ericstj** [comment]
This was fixed and ported to 9.0 - thank you @sobczyk and @carlossanlop
